scripts/ciTest.js

```diff
@@ -130,7 +130,7 @@ function runTests() {
         console.log(`testing ${file}`);
         // note existsSync test already ensure that it is a directory
         try {
-          cp.execSync(`node input.js`, { cwd: testDir, encoding: "utf8" });
+          cp.exec(`node input.js`, { cwd: testDir, encoding: "utf8" });
```

Comment:
If this is not sync anymore, doesn't that mean that success will be logged immediately and failure not caught?

Reply:
This was originally `exec` and @cristianoc changed it to sync in https://github.com/jchavarri/rescript-compiler/commit/0fbf257f0853520bd9f20c41fe9f49655135e632. I understood this was part of the fix for tests hanging, that we could safely revert now.

If a callback is passed to `exec` then [everything should work fine](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback):

> If a callback function is provided, it is called with the arguments (error, stdout, stderr). On success, error will be null. On error, error will be an instance of [Error](https://nodejs.org/api/errors.html#class-error). The error.code property will be the exit code of the process. By convention, any exit code other than 0 indicates an error. error.signal will be the signal that terminated the process.

Reply:
Indeed the call-back needs to be added back, or this won't fail on failed tests.

Reply:
Added back in https://github.com/rescript-lang/rescript-compiler/pull/5477/.